docs/extending/extending_client_side.md

```diff
@@ -0,0 +1,531 @@
+
+```javascript
+((Stimulus) => {
+    const StimulusExtra = Stimulus.start(); // start will instantiate the class and return its new instance
+    class SlugCheckController extends Stimulus.use.Controller {
+        compare() {
+            // noop
+        }
+        slugify() {
+            window.myCustomChecks.validate(this.element);
+        }
+        urlify() {
+            window.myCustomChecks.validate(this.element);
+        }
+    }
+    StimulusExtra.register('w-slug', SlugCheckController);
+})(window.Stimulus);
+```
+
+#### Extending an existing Stimulus controller
+
+For more extreme customisations it is possible to retrieve an existing registered Controller from the Stimulus application instance. This can be extended (using ES2015 class inheritance) and this extended controller registered again with the same identifier.
+
+There may also be some side effects of built in controllers being registered, depending on the timing of your JavaScript code event firing.
+
+The example below is a basic functional way to extend and override an existing controller.
+Stimulus' only documented way of retrieving a Controller constructor is via the [`getControllerForElementAndIdentifier`](https://stimulus.hotwired.dev/reference/controllers#directly-invoking-other-controllers) method on the application.
+You can also retrieve all registered Controllers via `window.Stimulus.controllers`, but this approach is not officially supported by Stimulus.
+
+Once you have the Controller class, you can use `window.Stimulus.unload` to remove the registration of the existing one and then replace it with your custom Controller class using `window.Stimulus.register`.
+
+Here is a basic example of this override.
+
+```javascript
+(() => {
+    const identifier = 'w-slug';
+    const element = document.querySelector(`[data-controller="${identifier}"]`);
+
+    if (!element) return;
+
+    const SlugController = window.Stimulus.getControllerForElementAndIdentifier(
+        element,
+        identifier,
+    );
+
+    class CustomSlugController extends SlugController {
+        slugify() {
+            // custom url slug
+        }
+        urlify() {
+            // custom url slug
+        }
+    }
+    Stimulus.register('w-slug', CustomSlugController);
+})();
+```
+
+#### Completely overriding existing admin behaviour of Stimulus controllers
+
+Wagtail also allows you to register a controller against its main application instance, see the examples above or the events reference for these events.
+
+You can completely override the built in controllers via using the same `identifier` (usually starts with `w-`) and registering your own controller with that identifier.
+
+It is important to note that your custom controller will need to re-implement the existing methods or you will get console errors when these are called.
+
+There may also be some side effects of built in controllers being registered, depending on the timing of your JavaScript code event firing.
```

Comment:
```suggestion
There may also be some side effects of built-in controllers being registered, depending on the timing of your JavaScript code event firing.
```

Side note: these two last parts seems a lot alike, I wonder if it is useful to keep both.